drivers/adodb-mssqlnative.inc.php

```diff
@@ -595,6 +605,12 @@ function _query($sql,$inputarr=false)
 
 		if ($this->debug) ADOConnection::outp("<hr>running query: ".var_export($sql,true)."<hr>input array: ".var_export($inputarr,true)."<hr>result: ".var_export($rez,true));
 
+		/*
+		* Store off the affected rows for retrieval later
+		*/
+		$this->sqlServerAffectedRows = sqlsrv_rows_affected($rez);
```

Comment:
According to [documentation](https://docs.microsoft.com/en-us/sql/connect/php/sqlsrv-query?view=sql-server-ver15#return-value), sqlsrv_query() returns _false_ if the statement cannot be created and/or executed.

I didn't test this, but I suppose that in such scenario, `sqlsrv_rows_affected($rez)` would throw some kind of error / warning.

Maybe better to do:
```php
if (!$rez) {
    $this->sqlServerAffectedRows = -1;
    return false;
} else {
    $this->sqlServerAffectedRows = sqlsrv_rows_affected($rez);
}
```